lumicks/pylake/scan.py

```diff
@@ -91,10 +91,137 @@ def _plot(self, image, frame=1, **kwargs):
             aspect=(image.shape[0] / image.shape[1]) * (x_um / y_um)
         )
 
-        plt.imshow(image, **{**default_kwargs, **kwargs})
+        if not image_handle:
+            image_handle = plt.imshow(image, **{**default_kwargs, **kwargs})
+        else:
+            # Updating the image data in an existing plot is a lot faster than re-plotting with `imshow`.
+            image_handle.set_data(image)
+
         plt.xlabel(f"{axis_label[axes[0]['axis']]} ($\\mu m$)")
         plt.ylabel(f"{axis_label[axes[1]['axis']]} ($\\mu m$)")
         if self.num_frames == 1:
             plt.title(self.name)
         else:
             plt.title(f"{self.name} [frame {frame}/{self.num_frames}]")
+
+        return image_handle
+
+    def _export_movie(self, plot_type, file_name, start_frame, end_frame, fps, **kwargs):
+        """Export a movie of a particular scan plot
+
+        Parameters
+        ----------
+        plot_type : str
+            One of the plot types "red", "green", "blue" or "rgb".
+        file_name : str
+            File name to export to.
+        start_frame : int
+            First frame in exported movie.
+        end_frame : int
+            Last frame in exported movie.
+        fps : int
+            Frame rate.
+        **kwargs
+            Forwarded to :func:`matplotlib.pyplot.imshow`."""
+        from matplotlib import animation, rcParams
+        import matplotlib.pyplot as plt
+
+        metadata = dict(title=self.name)
+        if "ffmpeg" in animation.writers:
+            writer = animation.writers['ffmpeg'](fps=fps, metadata=metadata)
+        elif "pillow" in animation.writers:
+            writer = animation.writers['pillow'](fps=fps, metadata=metadata)
+        else:
+            raise RuntimeError("You need either ffmpeg or pillow installed to export movies.")
+
+        start_frame = start_frame if start_frame else 1
+        end_frame = end_frame if end_frame else self.num_frames
+        end_frame += 1
+
+        # This prevents font anti-aliasing from being disabled on the notebook backends (at the cost of not having
+        # transparent regions outside the axes part of figures).
+        rcParams['figure.facecolor'] = "White"
```

Comment:
Indicate why it needs to be force-enabled. Workaround for some bug or some other reason? "Why" is one of the most important things to answer in comments and commit messages.

Reply:
Yeah. This ticket seems related and started my search for a solution in the realm of transparency: https://github.com/matplotlib/matplotlib/issues/5335
The actual tag `nbagg.transparent` has since then been deprecated.

I could make it a bit more specific (only override when there's transparency) and restore the color after export.

```python
        # On some notebook backends, figures render with a transparent background by default. This leads to very
        # poor image quality, since it prevents font anti-aliasing (at the cost of not having transparent regions
        # outside the axes part of figures).
        face_color = to_rgba(rcParams["figure.facecolor"])
        if face_color[3] < 1.0:
            rcParams["figure.facecolor"] = (face_color[0], face_color[1], face_color[2], 1.0)
        else:
            face_color = None
```

Reply:
It would be nice to include the link in the comment as well.